swupd/filesystem.go

```diff
@@ -79,9 +79,10 @@ func (m *Manifest) createManifestRecord(rootPath string, path string, fi os.File
 		fmt.Fprint(os.Stderr, err)
 	}
 
-	// remove leading "/" from manifest record
+	// the most recently added file is the one we need to update
 	f := m.Files[len(m.Files)-1]
-	f.Name = strings.TrimLeft(f.Name, "/")
+	// Only the bundle name should be part of the name in the manifest
+	f.Name = strings.Replace(f.Name, "/Manifest.", "", -1)
```

Comment:
This stuff screams "race condition" to me. As a general design would it not be better to have something that creates a record and a separate step that appends the record, which could then be done under lock? This function would then be "create record, hack the name, type and version fiends, append record". Of course we could just declare that Manifest types are not thread safe, and so can only be used from one go routine at a time.

Reply:
I see what you mean, yes I suppose appending should be done last. I was just trying to reuse code that already existed.